frontend/frame/ContentFactory.ts

```diff
@@ -0,0 +1,53 @@
+import { Content, ContentType } from '../../model/slide/content/Content';
+import { EmbedContentFactory } from './EmbedContentFactory';
+import { H5PContentFactory } from './H5PContentFactory';
+import { ImageContentFactory } from './ImageContentFactory';
+import { TextContentFactory } from './TextContentFactory';
+
+/**
+ * Base Class to convert JSON to Content objects
+ */
+export abstract class ContentFactory {
+  /**
+   * Converts JSON to Content Object
+   * @param json the JSON sent by the Webserver
```

Comment:
I would not specify that this json was sent by the webserver, since these factories function independently of the webserver